google-cloud-spanner/src/main/java/com/google/cloud/spanner/DatabaseClient.java

```diff
@@ -52,6 +53,35 @@
    */
   Timestamp write(Iterable<Mutation> mutations) throws SpannerException;
 
+  /**
+   * Writes the given mutations atomically to the database with the given options.
+   *
+   * <p>This method uses retries and replay protection internally, which means that the mutations
+   * are applied exactly once on success, or not at all if an error is returned, regardless of any
+   * failures in the underlying network. Note that if the call is cancelled or reaches deadline, it
+   * is not possible to know whether the mutations were applied without performing a subsequent
+   * database operation, but the mutations will have been applied at most once.
+   *
+   * <p>Example of blind write.
+   *
+   * <pre>{@code
+   * long singerId = my_singer_id;
+   * Mutation mutation = Mutation.newInsertBuilder("Singer")
+   *         .set("SingerId")
+   *         .to(singerId)
+   *         .set("FirstName")
+   *         .to("Billy")
+   *         .set("LastName")
+   *         .to("Joel")
+   *         .build();
+   * dbClient.writeWithOptions(Collections.singletonList(mutation));
+   * }</pre>
+   *
+   * @return a response with the timestamp at which the write was committed
+   */
+  CommitResponse writeWithOptions(Iterable<Mutation> mutations, TransactionOption... options)
```

Comment:
What do you think about `WriteOption` in place of `TransactionOption`?

Reply:
Thiago and I had a short discussion about this, and we think `TransactionOption` is better in this case, as we intend to also use this for options that we will need for normal read/write transactions.

Reply:
Ok sgtm